### PR TITLE
Added Interface Settings

### PR DESCRIPTION
**Pull request by Bhavya0304**
### Issue!
--> whenever someone try to use this script there were chances of error because Network Interface card name was hard coded so if the name of NIC is different on those computer script won't work 

### Fix
--> added a functionality to write network interface card name as command line argument!

### CODE REVIEW

File: deauthimposter.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
 #This just to check if system arguments are correct!!
 user_arg = sys.argv[1]
```
Comment (iamEzmuth) [inline deauthimposter.py:25]: Add conditions to check whether the args supplied or not!

File: deauthimposter.py [hunk 2]
```diff
@@ -9,7 +9,8 @@
...
-if(user_arg != "all" and user_arg != "select"):
+user_interface = sys.argv[2]
```
Comment (iamEzmuth) [inline deauthimposter.py]: Add conditions to check whether the args supplied or not!

File: Requires/premethods.sh [hunk 3]
```diff
@@ -1,7 +1,12 @@
 #!/bin/bash
 
-ifconfig wlan0 down
+if ["$1"]; then
```
Comment (iamEzmuth) [inline Requires/premethods.sh]: i am not sure this will work please ensure before committing

File: Requires/premethodsfix.sh [hunk 4]
```diff
@@ -1,9 +1,14 @@
 #!/bin/bash
 
-ifconfig wlan0 down
+if ["$1"]; then
```
Comment (iamEzmuth) [inline Requires/premethodsfix.sh]: i am not sure this will work please ensure before committing

File: Requires/premethods.sh [hunk 5]
```diff
@@ -1,7 +1,12 @@
 #!/bin/bash
 
-ifconfig wlan0 down
+if ["$1"]; then 
+    INTERFACE=$1
+else
+    INTERFACE="wlan0"    
+
```
Comment (iamEzmuth) [inline Requires/premethods.sh:9]: every if/else statements in shell scripting ends with fi

File: Requires/premethodsfix.sh [hunk 6]
```diff
@@ -1,9 +1,14 @@
 #!/bin/bash
 
-ifconfig wlan0 down
+if ["$1"]; then 
+    INTERFACE=$1
+else
+    INTERFACE="wlan0"    
+
```
Comment (iamEzmuth) [inline Requires/premethodsfix.sh:9]: every if/else statements in shell scripting ends with fi

**Review by iamEzmuth — CHANGES_REQUESTED** [review]
If you are really willing to contribute that's OK , first of all  i am so glad actually i created this script for my own sake but if you want to make it better that's great!

### Points to Focus: 

- If you are adding a new arg in main python file than update read-me file accordingly.
- I have reviewed the code and commented the issues. looks like you are new to contribution field no worries at all.
- Please verify and run the code on local machine before PR.
- if if if if and only if you are willing to help you can add a small section of help argument on main python file as you would have noticed in almost all Linux shell scripts (e.x --help) that provides basic arguments info. (If you are not willing to complete this final point its completely up-to you i will still approve and merge this PR )  

------BEST OF LUCK------

File: README.md [hunk 7]
```diff
@@ -1,36 +1,50 @@
 # DeAuthImposter
-Best tool to deauthenticate (disconnect) everyone in wifi network you are...
+Best tool to deauthenticate (disconnect) everyone in wifi network you want to...
 
-TO start the tool:
+## Installing dependencies
+```
+    sudo pip3 install -r Requirements.txt
```
Comment (Bhavya0304) [inline README.md:6]: added requirements file

**Review by iamEzmuth — APPROVED** [review]
Great Work!!! 
There still some fixes to solve but for now I am merging your request and will add those fixes to issues tab you can work on them if you want.

**Pull request merged**